Web/3DObject/src/WebAR3DObject.tsx

```diff
@@ -46,58 +71,78 @@ export function WebAR3DObject(props: WebAR3DObjectContainerProps): React.ReactEl
-            mxPositionXExpr={props.mxPositionXExpr}
-            mxPositionYExpr={props.mxPositionYExpr}
-            mxPositionZExpr={props.mxPositionZExpr}
-            mxRotationType={props.mxRotationType}
-            mxRotationXStat={props.mxRotationXStat}
-            mxRotationYStat={props.mxRotationYStat}
-            mxRotationZStat={props.mxRotationZStat}
-            mxRotationXAtt={props.mxRotationXAtt}
-            mxRotationYAtt={props.mxRotationYAtt}
-            mxRotationZAtt={props.mxRotationZAtt}
-            mxRotationXExpr={props.mxRotationXExpr}
-            mxRotationYExpr={props.mxRotationYExpr}
-            mxRotationZExpr={props.mxRotationZExpr}
-            mxScaleType={props.mxScaleType}
-            mxScaleXStat={props.mxScaleXStat}
-            mxScaleYStat={props.mxScaleYStat}
-            mxScaleZStat={props.mxScaleZStat}
-            mxScaleXAtt={props.mxScaleXAtt}
-            mxScaleYAtt={props.mxScaleYAtt}
-            mxScaleZAtt={props.mxScaleZAtt}
-            mxScaleXExpr={props.mxScaleXExpr}
-            mxScaleYExpr={props.mxScaleYExpr}
-            mxScaleZExpr={props.mxScaleZExpr}
-            OnSceneLoaded={handleSceneLoaded}
-            mxMaterialColor={props.mxMaterialColor.value ?? "#0CABF9"}
-            mxMaterialOption={props.mxMaterialOption}
-            texture={texture}
-            mxMetalness={props.mxMetalness}
-            mxOpacity={props.mxOpacity}
-            mxRoughness={props.mxRoughness}
-            mxLightingType={props.mxLightingType}
-            mxUseDraggingInteraction={props.mxUseDraggingInteraction}
-            mxUsePinchInteraction={props.mxUsePinchInteraction}
-            mxDraggingEnabled={props.mxDraggingEnabled.value ?? false}
-            mxDragType={props.mxDragType}
-            mxOnDrag={props.mxOnDrag}
-            mxPinchEnabled={props.mxPinchEnabled.value ?? false}
-            mxPinchToScaleEnabled={props.mxPinchToScaleEnabled}
-            mxOnPinchActionValue={props.mxOnPinchActionValue}
-            mxOnHoverEnter={props.mxOnHoverEnter}
-            mxOnHoverExit={props.mxOnHoverExit}
-            mxOnClick={props.mxOnClick}
-        />
+        <>
+            <GizmoComponent
+                color={props.mxGizmoColor?.value ?? "#ffffff"}
+                mesh={rotationSet ? rootMesh : undefined}
+                gizmoSize={Number(props.mxGizmoSize.value) ?? 0.05}
+                draggingEnabled={props.mxDraggingEnabled.value ?? false}
+                pinchEnabled={props.mxPinchEnabled.value ?? false}
+                rotationEnabled={props.mxPinchRotationEnabled.value ?? false}
+                onScale={newScale => {
+                    setScale(newScale);
+                }}
+                onDrag={newPosition => {
+                    setPosition(newPosition);
+                }}
+                onRotate={newRotation => {
+                    setRotation(newRotation);
+                }}
+            />
```

Comment:
Looking at this place I'm wondering -- why we need a component???
It's look like it's perfect case for hook